### PR TITLE
dependencies: bump `go-azure-sdk` to `v0.20230508.1101604`

### PR DESCRIPTION
**Pull request by stephybun**
Also migrating the following services to the new base layer:

- `digitaltwins`
- `elastic`
- `eventhub`
- `healthcare`
- `iotcentral`
- `kubernetesconfiguration`
- `search`

### CODE REVIEW

File: internal/services/containers/client/client_gen.go [hunk 1]
```diff
@@ -6,6 +6,9 @@ import (
 	"github.com/hashicorp/terraform-provider-azurerm/internal/common"
 )
 
+type autoClient struct {
+}
+
```
Comment (tombuildsstuff) [inline internal/services/containers/client/client_gen.go:11]: this file's generated so this'll get reverted?

```suggestion
```

File: internal/services/containers/client/client.go [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package client
 
 import (
+	"fmt"
 	"github.com/Azure/go-autorest/autorest/azure"
```
Comment (tombuildsstuff) [inline internal/services/containers/client/client.go:5]: imports?

**Review by tombuildsstuff — APPROVED** [review]
A couple of minor comments but otherwise 👍

File: internal/services/containers/client/client_gen.go (hunk 1 above)
Comment (stephybun) [reply internal/services/containers/client/client_gen.go:11]: whoops this is from some other work

**Pull request merged**